oauth2_http/javatests/com/google/auth/oauth2/MockMetadataServerTransport.java

```diff
@@ -271,14 +275,14 @@ public LowLevelHttpResponse execute() {
     if (queryPairs.containsKey("format")) {
       if (((String) queryPairs.get("format")).equals("full")) {
 
-        // return license only if format=full is set
-        if (queryPairs.containsKey("license")) {
-          if (((String) queryPairs.get("license")).equals("TRUE")) {
+        // return licenses only if format=full is set
+        if (queryPairs.containsKey("licenses")) {
+          if (BOOL_PARAMETER_VALUE.matcher((String) queryPairs.get("licenses")).matches()) {
```

Comment:
qq, why is `BOOL_PARAMETER_VALUE` Pattern needed here?

Reply:
@lqiu96 sorry for the wait! `BOOL_PARAMETER_VALUE` captures all values that the GCE metadata server accepts as true for a boolean query component (numeric: `1`, case sensitive: `on`, case insensitive: `y`, `yes`, `true`) or false if unmatched. I only added it to mock the behavior of the GCE metadata server accurately even though the library will only send `TRUE` in a request.

Reply:
Ah, ok. Would you be able link to any documentation on the boolean query values that MDS accepts (on,y, etc.)? If it doesn't exist, could you add a small comment to explain the above for future maintainers?

Reply:
The MDS calls `::cloud_cluster::metadata_parsing_utils::GetBoolQueryComponentDefault`. I can leave a `//google3/...` reference here, but that wouldn't be helpful to external users.
My suggestion would be to reference the Abseil docs since `GetBoolQueryComponentDefault` is (currently) just `::absl::SimpleAtob` extended with matches for `on` and `off`.

```suggestion
          // The metadata server defaults to false and matches "on", "off" and ::absl::SimpleAtob.
          // See https://abseil.io/docs/cpp/guides/strings#numericConversion for more information.
          if (BOOL_PARAMETER_VALUE.matcher((String) queryPairs.get("licenses")).matches()) {
```